### PR TITLE
CORE-6411 Add Jenkinsfile 

### PR DESCRIPTION
**Pull request by E1P3**
Ensure the release is automated using shared library pipeline

Env var names changed to match what the Jenkins pipeline expects and the repo name set dynamically from there

Jenkins pipeline set up which will only Build tags

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -20,7 +20,7 @@ String getPackageNames(File file) {
 
 subprojects { sub ->
     group 'net.corda.kotlin'
-    version kotlin_version
+    version="00-test"+kotlin_version
```
Comment (chrisr3) [inline build.gradle]: This doesn't look right for the "main" branch.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+@Library('corda-shared-build-pipeline-steps@CORE-6347/modifications-for-corda-gradle-plugins') _
+
+cordaPipeline(
+    publishRepoPrefix: 'corda-dependencies-dev',
+    slimBuild: true,
+    runUnitTests:false,
+    )
```
Comment (chrisr3) [inline Jenkinsfile]: For testing only?

**Review by chrisr3 — COMMENTED** [review]
This cannot possibly be the final version?!

File: build.gradle (hunk 1 above)
Comment (ronanbrowne) [reply build.gradle]: @E1P3 +1 revert to original,  This was just for testing purposes.

File: Jenkinsfile (hunk 2 above)
Comment (ronanbrowne) [reply Jenkinsfile]: yes change to corda-dependencies please @E1P3

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,8 @@
+@Library('corda-shared-build-pipeline-steps@CORE-6347/modifications-for-corda-gradle-plugins') _
```
Comment (ronanbrowne) [inline Jenkinsfile]: This needs to point to 5.0 ( we wont merge till accompanying changes approved)

**Review by ronanbrowne — CHANGES_REQUESTED** [review]

**Review by chrisr3 — APPROVED** [review]
LGTM

**Review by ronanbrowne — APPROVED** [review]
LGTM

**Pull request merged**